src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/IndexUsage.java

```diff
@@ -0,0 +1,34 @@
+package io.stargate.sgv2.jsonapi.service.cqldriver.executor;
+
+import io.micrometer.core.instrument.Tags;
+
+public interface IndexUsage {
+
+  /** Default implementation when we are not tracking index usage for the schema object */
+  IndexUsage NO_OP =
+      new IndexUsage() {
+        @Override
+        public Tags getTags() {
+          return Tags.empty();
+        }
+
+        @Override
+        public void merge(IndexUsage indexUsage) {
+          // NO-OP
+        }
+      };
+
+  /**
+   * This method is used to generate the tags for the index usage
+   *
+   * @return
+   */
+  Tags getTags();
+
+  /**
+   * This method is used to merge the index usage of two different types for filters used in a query
```

Comment:
Should indicate what is being updated; I assume `this` is updated with tags from `indexUsage`